biocrnpyler/dna_construct.py

```diff
@@ -505,15 +751,16 @@ def __init__(self,
                 attributes=None,
                 initial_conc=None,
                 copy_parts=True,
+                component_enumerators = (TxExplorer(),),
```

Comment:
I'm a bit confused by this. Tuples are immutable, so no new elements can **not** be added to the tuple. Is there a reason for such a construct? 
```python
component_enumerators = [TxExplorer()]
```
Consider using a  list instead

Reply:
if i do it that way then your check for mutable default arguments goes off and the code fails. I could also make it none and set it in the body of init (if none) but this makes it more obvious what the default value is

Reply:
I did this because we are not allowed to use mutable lists in the default arguments. if you want to add elements you would pass in a list with more elements

Reply:
I would put this in the constructor of DNA_Assembly if None.

Reply:
The usual way to do this is something like this:
```python
def some_function(some_other_argument,component_enumerators=None):
   if component_enumerators is None:
      component_enumerators=(TxExplorer(),),)
```
You cannot enforce the structure or content of a list as the default argument of a function. But you can hint at it with type hints. For example:
```python
def some_function(some_arguemnt,component_enumerators: List[TxTlExplorer_CE]=None):
```
This serves as a type hint for the users, what kind of list is needed as an input to the function.